packages/data-context/src/actions/DataEmitterActions.ts

```diff
@@ -1,7 +1,34 @@
+import pDefer from 'p-defer'
+import { EventEmitter } from 'stream'
+
 import type { DataContext } from '../DataContext'
 
-export class DataEmitterActions {
-  constructor (private ctx: DataContext) {}
+abstract class DataEmitterEvents {
+  protected pub = new EventEmitter()
```

Comment:
Not a blocker but what do you think about using native JS features like `#` for private members instead of compile-time-only features like `protected`?

This would become

```ts
abstract class DataEmitterEvents {
  #pub = new EventEmitter()
}
```

No practical difference, but one leverages the actual runtime language primitives, and one uses a proprietary, compile-time only keyword. Any benefit in using `protected` here? Seems `#` is generally going to better in any given scenario.

Reply:
Yeah, it's protected because it's used in `subscribeTo`, which is located on the `DataEmitterActions` class as well as `_emit`, which is in the `DataEmitterEvents`. I'm fine with moving to `#` for things that are actually private though